File: HubWebApp/Internal/XTI_Hub/InstallConfiguration.cs
using XTI_App.Abstractions;
using XTI_Hub.Abstractions;
using XTI_HubDB.Entities;

namespace XTI_Hub;

public sealed class InstallConfiguration
{
    private readonly HubFactory hubFactory;
    private readonly InstallConfigurationEntity config;

    internal InstallConfiguration(HubFactory hubFactory, InstallConfigurationEntity config)
    {
        this.hubFactory = hubFactory;
        this.config = config;
    }

    public bool IsFound() => config.ID > 0;

    internal Task Update(InstallConfigurationTemplate template, int installSequence, CancellationToken ct) =>
        hubFactory.DB.InstallConfigurations.Update
        (
            config,
            c =>
            {
                c.TemplateID = template.ID;
                c.InstallSequence = installSequence;
            },
            ct
        );

    internal Task Delete(CancellationToken ct) =>
        hubFactory.DB.InstallConfigurations.Delete(config, ct);

    public async Task<InstallConfigurationModel> ToModel(CancellationToken ct)
    {
        var template = await hubFactory.InstallConfigurationTemplates.Template(config.TemplateID, ct);
        return new
        (
            ID: config.ID,
            ConfigurationName: config.ConfigurationName,
            AppKey: new AppKey(new AppName(config.AppName), AppType.Values.Value(config.AppType)),
            Template: template.ToModel(),
            InstallSequence: config.InstallSequence
        );
    }

}

File: HubWebApp/Apps/HubWebApp/Scripts/Lib/InstallConfiguration.ts
import { AppKey } from "./AppKey";
import { InstallConfigurationTemplate } from "./InstallConfigurationTemplate";

export class InstallConfiguration {
    readonly id: number;
    readonly repoOwner: string;
    readonly repoName: string;
    readonly configurationName: string;
    readonly appKey: AppKey;
    readonly template: InstallConfigurationTemplate;
    readonly installSequence: number;

    constructor(source?: IInstallConfigurationModel) {
        this.id = source?.ID || 0;
        this.repoOwner = source?.RepoOwner || "";
        this.repoName = source?.RepoName || "";
        this.configurationName = source?.ConfigurationName || "";
        this.appKey = new AppKey(source && source.AppKey);
        this.template = new InstallConfigurationTemplate(source && source.Template);
        this.installSequence = source?.InstallSequence || 0;
    }

    get isFound() { return this.id > 0; }
}